common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1572,3 +1572,11 @@ def set_user_partitions_for_scheme(self, partitions, scheme):
             if p.scheme != scheme
         ]
         self.user_partitions = other_partitions + partitions  # pylint: disable=attribute-defined-outside-init
+
+    def is_discussion_enabled(self, user):
+        """
+        Return True if Discussion is enabled for a course; else False
+        """
+        if not any([tab.type == 'discussion' and tab.is_enabled(self, user) for tab in self.tabs]):
```

Comment:
@wajeeha-khalid The `CourseTabList` inherits from the xBlock `Field` class.  We explicitly chose for `course_overviews` to not expose the xBlock interface, in order to limit coupling across the interfaces/implementation.  If there is any common functionality that needs to be shared across `course_overviews` and `xBlocks`, we would refactor the code and create a common utility function instead.

So in this case, we want to cache and store only the list of tab names in `course_overviews`.